db/httpdb.js

```diff
@@ -439,6 +440,52 @@ module.exports = function (
     )
   }
 
+  DB.prototype.lockAccount = function (account) {
+    var unlockCode = crypto.randomBytes(16).toString('hex');
+    log.trace({ op: 'DB.lockAccount', uid: account && account.uid, unlockCode: unlockCode })
+
+    return this.pool.post(
+      '/account/' + account.uid.toString('hex') + '/lock',
+      {
+        lockedAt: Date.now(),
+        unlockCode: unlockCode
+      }
+    )
+  }
+
+  DB.prototype.unlockAccount = function (account) {
+    log.trace({ op: 'DB.unlockAccount', uid: account && account.uid })
+    return this.pool.post(
+      '/account/' + account.uid.toString('hex') + '/unlock'
+      )
+      .then(null,
+        function (err) {
+          if (err.statusCode === 404) {
+            err = error.unknownAccount()
+          }
+          throw err
+        }
+      )
+  }
+
+  DB.prototype.unlockCode = function (account) {
+    log.trace({ op: 'DB.unlockCode', uid: account && account.uid })
+    return this.pool.get(
+      '/account/' + account.uid.toString('hex') + '/unlockCode'
+      )
+      .then(
+        function (body) {
+          return body.unlockCode
```

Comment:
We deal with buffers and hex strings a lot. To help not get them mixed up we try to keep them buffers "internally" and decode/encode them on the edges. When we send data out we toString them and create buffers as soon as we can when we receive them. So here we should do something like `return bufferize(body).unlockCode`. Then in `accounts.js` we won't need to worry about converting it to a buffer.